src/tree/azure/GroupingItem.ts

```diff
@@ -47,63 +50,103 @@ export class GroupingItem implements ResourceGroupsItem {
             this.portalUrl = createPortalUrl(resourceGroup.subscription, resourceGroup.id);
         }
 
-        this.subscription = {
+        this.subscription = this.context ? {
             // for v1.5 compatibility
             ...this.context.subscriptionContext,
             ...this.context.subscription,
-        };
+        } : undefined;
+
+        if (this.resourceGroup) {
+            this.id = this.resourceGroup.id;
+        } else {
+            if (this.context?.subscription) {
+                this.id = `/subscriptions/${this.context?.subscriptionContext.subscriptionId}/groupings/${this.label}`;
+            } else {
+                this.id = `/groupings/${this.label}`;
+            }
+        }
     }
 
     // Needed for context menu commands on the group tree items. E.g. "Create..."
-    public readonly subscription: ISubscriptionContext & AzureSubscription;
+    public readonly subscription?: ISubscriptionContext & AzureSubscription;
 
-    readonly id: string = this.resourceGroup ? this.resourceGroup.id : `/subscriptions/${this.context.subscriptionContext.subscriptionId}/groupings/${this.label}`;
+    readonly id: string;
 
     async getChildren(): Promise<ResourceGroupsItem[] | undefined> {
-        const sortedResources = this.resources.sort((a, b) => a.name.localeCompare(b.name));
+
+        const sortedResources = this.resources.sort((a, b) => {
+
+            if (a.subscription !== b.subscription) {
+                return a.subscription.name.localeCompare(b.subscription.name);
+            }
+
+            return a.name.localeCompare(b.name);
+        });
+
+        const groupMap = new Map<AzureSubscription, AzureResource[]>();
```

Comment:
Why do we need this groupMap now? Is it so we don't have repeated resource IDs in sorted resources?

Reply:
The purpose of this map is to add in the subscription separators (description-only tree items) if needed.

Reply:
I think it's worth writing a comment or changing the variable name to something more descriptive.

Like `subscriptionGroupingMap` or `resourcesGroupedBySubscriptionMap`